### PR TITLE
duty day script

### PR DESCRIPTION
**Pull request by idelcano**
https://app.clickup.com/t/865ct8tva

docker pull docker.eyeseetea.com/widpit/alpineupdated:latest

docker run -it --privileged --rm -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix --mount src="$(pwd)",target=/test_container,type=bind -p 8085:8080 docker.eyeseetea.com/widpit/alpineupdated

### CODE REVIEW

**Review by ifoche — CHANGES_REQUESTED** [review]
thanks @idelcano but could you please remove hostnames from it? config file you could put it as an example without any information and then we keep the actual file private

File: DHIS2/dutyday/config.json
Comment (nshandra) [inline DHIS2/dutyday/config.json:1]: Maybe monit and diskspace can be merged in a single entry.

File: DHIS2/dutyday/logger.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#this file must be placed in the instances
+set +x
+
+monitlogger() {
+  echo ""
+	echo "----------monitlog----------"
+	echo
+	sudo monit status | awk '/Process/{p=$0} /Not monitored/{print p "\n" $0 "\n"}'
+	echo
+}
```
Comment (nshandra) [inline DHIS2/dutyday/logger.sh:11]: The value of "monit" field could be used here. That way, if we needed to change the command there would be no need to redeploy logger.sh.

In the other hand that's probably unlikely.

File: DHIS2/dutyday/reporter.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    client.connect(host, username="tomcatuser", pkey=private_key)
+
+    stdin, stdout, stderr = client.exec_command(command)
+    output = stdout.read().decode().strip()
+
+    client.close()
+
+    return output
+
```
Comment (nshandra) [inline DHIS2/dutyday/reporter.py:25]: I get why its hard-coded in the script, as we both use the same docker image, but in a generic case the key path and the username should be either command line arguments or in the config.json file.

Given the intended users of the script its fine like this.

File: DHIS2/dutyday/reporter.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+def runlogger(file):
+    logger = ""
+    with open(file, 'r') as file:
+        data = json.load(file)
+        for item in data:
+            logger = logger + "\n"
+            logger = logger + "Type: {"+item['type']+"}, Server: {"+item['server']+"}"
+            logger = logger + "\n"
+            # Accede a el Accede a elementos opcionales de manera segura
```
Comment (nshandra) [inline DHIS2/dutyday/reporter.py:68]: Typo?
Also, change to English

File: DHIS2/dutyday/logger.sh [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    if [ $type == "docker" ];then
+      dockerid=$(docker ps | grep $host"-core" | awk '{print $1}')
+
+      response=$(docker cp $dockerid +":"+$file+" /backup/logs/.")
+      file=$(basename $file)
+      LOG_FILE="/backup/logs/$file"
+    fi
+    DAYS_BACK=2
+
+    # Calcula la fecha de inicio basándose en los días hacia atrás
```
Comment (nshandra) [inline DHIS2/dutyday/logger.sh:61]: Change to English

File: DHIS2/dutyday/reporter.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import sys
+import subprocess
+import json
+
+import paramiko
+
+
+#this script will be executed on local docker with vpn active
+def execute_command_on_remote_machine(host, command):
+    path_to_private_key = '/root/.ssh/id_rsa'  # ruta a tu archivo de clave privada SSH
```
Comment (nshandra) [inline DHIS2/dutyday/reporter.py:12]: change comment to English

**Review by nshandra — CHANGES_REQUESTED** [review]
The script works and its a great tool. We both have used it so we know it works.

I have taken the liberty of adding a readme and making the changes to `config.json` requested by @ifoche.

My comments are mainly suggestions to consider implementing if there is some new feature added.

As for the requested changes:

- fix the typo
- rewrite the Spanish comments to English
- Check my `README.md` in case i missed something

All in all, good work.

**Pull request merged**